cdp_scrapers/scraper_utils.py

```diff
@@ -66,6 +80,352 @@ def str_simplified(input_str: str) -> str:
+        Set roles[i].title as "Council President" if match
+        and roles[i].body is a primary body like City Council
+    chair_patterns: List[str]
+        Set roles[i].title as "Chair" if match
+        and roles[i].body is not a primary body
+
+    Notes
+    -----
+    Remove roles[#] if roles[#].body in static_data.primary_bodies.
+    Use static_data.persons[#].seat.roles instead.
+
+    If roles[i].body not in static_data.primary_bodies,
+    roles[i].title cannot be "Councilmember" or "Council President".
+
+    Use "City Council" and "Council Briefing"
+    if static_data.primary_bodies is empty.
+    """
+    if roles is None:
+        roles = []
+
+    if not static_data or not static_data.primary_bodies:
+        # Primary/full council not defined in static data file.
+        # these are reasonably good defaults for most municipalities.
+        primary_body_names = ["city council", "council briefing"]
+    else:
+        primary_body_names = [
+            body_name.lower() for body_name in static_data.primary_bodies.keys()
+        ]
+
+    try:
+        have_primary_roles = len(static_data.persons[person_name].seat.roles) > 0
+    except (KeyError, AttributeError, TypeError):
+        have_primary_roles = False
+
+    def _is_role_period_ok(role: Role) -> bool:
+        """
+        Test that role.[start | end]_datetime is acceptable
+        """
+        if role.start_datetime is None or role.end_datetime is None:
+            return False
+        if not have_primary_roles:
+            # no roles in static data; accept if this this role is current
+            return role.start_datetime.astimezone(
+                pytz.utc
+            ) <= datetime.today().astimezone(pytz.utc) and datetime.today().astimezone(
+                pytz.utc
+            ) <= role.end_datetime.astimezone(
+                pytz.utc
+            )
+        # accept if role coincides with one given in static data
+        for static_role in static_data.persons[person_name].seat.roles:
+            if (
+                static_role.start_datetime <= role.start_datetime
+                and role.end_datetime <= static_role.end_datetime
+            ):
+                return True
+        return False
+
+    def _is_primary_body(role: Role) -> bool:
+        """
+        Is role.body one of primary_bodies in static data file
+        """
+        return (
+            role.body is not None
+            and role.body.name is not None
+            and str_simplified(role.body.name).lower() in primary_body_names
+        )
+
+    def _fix_primary_title(role: Role) -> str:
+        """
+        Council president or Councilmember
+        """
+        if (
+            role.title is None
+            or re.search(
+                "|".join(council_pres_patterns), str_simplified(role.title), re.I
+            )
+            is None
+        ):
+            return RoleTitle.COUNCILMEMBER
+        return RoleTitle.COUNCILPRESIDENT
```

Comment:
I think this is a bug. Should you really be looking for either and saying member?

Reply:
Or whatever if neither is found? We default return president?

Reply:
I think I just misunderstood this func. I get it.